### PR TITLE
nixos/iwd: add settings option

### PR DESCRIPTION
**Pull request by moni-dz**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/networking/iwd.nix [hunk 1]
```diff
@@ -5,7 +5,28 @@ with lib;
 let
   cfg = config.networking.wireless.iwd;
 in {
-  options.networking.wireless.iwd.enable = mkEnableOption "iwd";
+  options.networking.wireless.iwd = {
+    enable = mkEnableOption "iwd";
+
+    settings = mkOption {
+      type = (pkgs.formats.ini {}).type;
```
Comment (aanderse) [inline nixos/modules/services/networking/iwd.nix:12]: Maybe add `pkgs.formats.ini {}` in the `let` statement. 🤷‍♂️

File: nixos/modules/services/networking/iwd.nix [hunk 2]
```diff
@@ -5,7 +5,28 @@ with lib;
...
+
+        Network = {
+          EnableIPv6 = true;
+          RoutePriorityOffset = 300;
+        };
+      };
+
+      description = ''
+        Options passed to iwd.
+        See https://iwd.wiki.kernel.org/networkconfigurationsettings for supported options.
```
Comment (aanderse) [inline nixos/modules/services/networking/iwd.nix:26]: `<xlink` would make this link clickable.

File: nixos/modules/services/networking/iwd.nix [hunk 3]
```diff
@@ -15,6 +36,8 @@ in {
       '';
     }];
 
+    environment.etc."iwd/main.conf".source = (pkgs.formats.ini {}).generate "main.conf" cfg.settings;
```
Comment (aanderse) [inline nixos/modules/services/networking/iwd.nix:39]: Do we want to restart the service whenever this file changes, or leave it to the user to manually restart the service when convenient?

Comment (fortuneteller2k) [reply nixos/modules/services/networking/iwd.nix]: I would want to restart whenever the file changes, so I added a `restartTriggers` entry.

**Comment by aanderse** [comment]
@fortuneteller2k I'm sorry no one else came around to review this. I think this change is simple enough we can merge it with only my review. When you get a chance can you please rebase and ping me, then I'll merge. Thank you!

**Comment by fortuneteller2k** [comment]
@aanderse rebased

**Pull request merged**